### ISSUE TITLE
Fix end time of app start

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#3461]
### Description

Currently, the end of the app start is when the OS posts `UIWindowDidBecomeVisibleNotification`, which doesn't mean that some proper content is rendered for the user. When looking at an app start transaction, we can see that the initial frame render ends before the initial display span even starts. 
![CleanShot 2023-11-29 at 15 49 11](https://github.com/getsentry/sentry-cocoa/assets/2443292/2f3d8e9d-53e1-4d7a-a349-31711e76b32d)

Instead, the end of the initial frame render span and the app start should be the same as of the end of TTID. The SDK reports TTID when the app draws the next frame after `loadView` of the UIViewController is called, which correctly reflects when the user sees the first frame of proper content.

**Comment by armcknight** [comment]
Is this example reporting that TTID == TTFD? It's a bit confusing because I expect TTFD to always come after TTID, even if they're equal, but that's not what the frontend is showing here.

**Comment by philipphofmann** [comment]
No TTFD is shortly after TTID. You should call `SentrySDK.reportFullyDisplayed` once your UIViewController has done all the heavy lifting. This might occur before TTID because we report TTID once the next frame is drawn. If you have a UIViewController doing very little, and you manually call `SentrySDK.reportFullyDisplayed`, that can happen. I also see this in our iOS-Swift sample app. We could highlight this in the product to state that this view was rendered on time.

**Comment by armcknight** [comment]
> TTFD is shortly after TTID
Definitely seems like there's a bug in the frontend since in your screenshot it shows TTFD and then TTID. That should never happen IIUC.

**Comment by philipphofmann** [comment]
Ah you are talking about this ?

![CleanShot 2023-12-01 at 10 35 46@2x](https://github.com/getsentry/sentry-cocoa/assets/2443292/0c42cc31-8c84-4ed0-9db9-0847e8b6b132)

Yes, that seems like a bug.

**Comment by armcknight** [comment]
I'm still a bit unclear as to what we expect to see in the screenshot. It seems like TTFD should be at the end of `ui.load.full_display`, is that right?

This might be breaking in the sense that the aggregates are going to be weird once a customer goes live with this update, but they will even out after some time. I'm not sure that's really a breaking change w.r.t. the SDK, this is a bugfix because we weren't correctly reporting the timestamps, IIUC.

If it really is the case that we are currently misreporting times, I don't think we should gate the change behind what is effectively a feature flag, and force the customer to consider the tradeoff of possibly continuing with bad data. That makes configuration more complicated for customers, and code maintenance and customer support more complicated for us.

**Comment by philipphofmann** [comment]
@armcknight, I agree. We will fix the TTID/TTFD spans and measurements in a non-breaking change without a feature flag in https://github.com/getsentry/sentry-cocoa/issues/3471.

I'm still unsure if the app start duration change is breaking. In our weekly sync, we (@brustolin, @kahest, and me) agreed that it is. The plan for this issue is to prolong the initial frame render span until the next frame is drawn after `UIWindowDidBecomeVisibleNotification` was posted, which will extend app start duration statistics, but it's the right thing to do. The downside is that this could confuse users a bit.

**Comment by philipphofmann** [comment]
I made this change in https://github.com/getsentry/sentry-cocoa/pull/3530 in a non-breaking way behind a feature flag, `performanceV2`. When we update to the next major, we only need to make https://github.com/getsentry/sentry-cocoa/pull/3530 the default behavior.

**Comment by philprime** [comment]
Closing — this was fixed by #3530 (merged Jan 2024). App start end time is now based on the first frame drawn via `SentryFramesTrackerListener`, not `UIWindowDidBecomeVisibleNotification`. The `performanceV2` feature flag that originally gated this has been removed — first-frame measurement is now the default.
